spec/spec.md

```diff
@@ -526,81 +729,245 @@ response when returning `anydata` directly from a resource function.
+
+Supported From Ballerina - :heavy_check_mark:     
+Supported From Azure - :white_check_mark:
+
+| Type                	 | Trigger            	 | Input              	 | Output             	 |
+|-----------------------|----------------------|----------------------|----------------------|
+| Blob storage        	 | :heavy_check_mark: 	 | :heavy_check_mark: 	 | :heavy_check_mark: 	 |
+| Azure Cosmos DB     	 | :heavy_check_mark: 	 | :heavy_check_mark: 	 | :heavy_check_mark: 	 |
+| Azure SQL (preview) 	 | 	                    | :white_check_mark: 	 | :white_check_mark: 	 |
+| Dapr                	 | 	                    | :white_check_mark: 	 | :white_check_mark: 	 |
+| Event Grid          	 | :white_check_mark: 	 | 	                    | :white_check_mark: 	 |
+| Event Hubs          	 | :white_check_mark: 	 | 	                    | :white_check_mark: 	 |
+| HTTP & webhooks     	 | :heavy_check_mark: 	 | 	                    | :heavy_check_mark: 	 |
+| IoT Hub             	 | :white_check_mark: 	 | 	                    | 	                    |
+| Kafka               	 | :white_check_mark: 	 | 	                    | :white_check_mark: 	 |
+| Mobile Apps         	 | 	                    | :white_check_mark: 	 | :white_check_mark: 	 |
+| Notification Hubs   	 | 	                    | 	                    | :white_check_mark: 	 |
+| Queue storage       	 | :heavy_check_mark: 	 | 	                    | :heavy_check_mark: 	 |
+| RabbitMQ            	 | :white_check_mark: 	 | 	                    | :white_check_mark: 	 |
+| SendGrid            	 | 	                    | 	                    | :white_check_mark: 	 |
+| Service Bus         	 | :white_check_mark: 	 | 	                    | :white_check_mark: 	 |
+| SignalR             	 | :white_check_mark: 	 | :white_check_mark: 	 | :white_check_mark: 	 |
+| Table storage       	 | 	                    | :white_check_mark: 	 | :white_check_mark: 	 |
+| Timer               	 | :heavy_check_mark: 	 | 	                    | 	                    |
+| Twilio              	 | 	                    | 	                    | :heavy_check_mark: 	 |
+
+
+### 3.1. Triggers
+[Triggers](https://learn.microsoft.com/en-us/azure/azure-functions/functions-triggers-bindings?tabs=csharp) are the starting point of functions. They cause functions to invoke. A function must have only one trigger. This is the motivation for listener-trigger mapping. Currently, Ballerina supports 5 triggers.
+#### 3.1.1. HttpTrigger
+[HttpTrigger](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=in-process%2Cfunctionsv2&pivots=programming-language-java) is something that can be invoked with an HTTP request. This can be defined as follows in Ballerina Azure Function.
+```ballerina
+service /hello on new af:HttpListener() {
+    resource function post foo/[string bar](@http:Payload string greeting,) returns @af:HttpOutput string  {
```

Comment:
Extra comma in parameter. 
Also, shall we use json or a record instead of string?  Make sure to add the content type json for the curl command too